Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp

```diff
@@ -224,7 +226,8 @@ View::View(struct wpe_view_backend* backend, WPEDisplay* display, const API::Pag
                 return TRUE;
             }
             case WPE_EVENT_TOUCH_MOVE:
-                // FIXME: gestures
+                if (auto gestureController = wpe_view_get_gesture_controller(webView.wpeView()))
+                    webView.handleGesture(wpe_gesture_controller_detect_gesture(gestureController, event), *gestureController);
```

Comment:
I think it would be simpler if make `handleGesture()` receive only the event, call it unconditionally in the three cases and handle everything there.